shared/simplestreams/products.go

```diff
@@ -271,7 +279,9 @@ func (s *Products) ToLXD() ([]api.Image, map[string][][]string) {
 					}
 				}
 
-				if item.FileType == "lxd.tar.xz" {
+				if !lxdCompatFound && shared.StringInSlice(item.FileType, []string{"lxd.tar.xz", "incus.tar.xz"}) {
+					lxdCompatFound = true // Stop at first compatible item found.
```

Comment:
For my understanding, why should we stop at the first compatible item ?

Reply:
Because we only need to find a single compatible item, right now the `images:` remote is listing both for compat with current and older versions of LXD, but that wont always be the case:

https://uk.lxd.images.canonical.com/streams/v1/images.json

![image](https://github.com/canonical/lxd/assets/280181/fd170360-7769-4002-9d2e-45b008159ad5)

Reply:
thanks